File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.swift
import Foundation

let nmv = readLine()!.split(separator: " ").map{Int(String($0))!}
var edges:[[Int]] = []
for _ in 0..<nmv[1] {
    edges.append(readLine()!.split(separator: " ").map{Int(String($0))!})
}
var graph:[[Int]] = Array(repeating: [], count: nmv[0]+1)
var ch = Array(repeating: false, count: nmv[0]+1) // 정점의 방문여부 체크 배열

for i in 0..<nmv[1] {
    let l = edges[i][0]
    let r = edges[i][1]
    
    graph[l].append(r)
    graph[r].append(l)

    graph[l].sort()
    graph[r].sort()
}

var answer = ""
dfs(nmv[2])
print(answer)

answer = ""
ch = Array(repeating: false, count: nmv[0]+1)
bfs(nmv[2])
print(answer)

func dfs(_ n: Int) -> Void {
    ch[n] = true
    answer += "\(n) "
    for c in graph[n] {
        if !ch[c] { dfs(c) }
    } 
}

func bfs(_ n: Int) -> Void {
    var queue = [n]
    
    while(queue.count > 0) {
        let p = queue.removeFirst()
        if !ch[p] {
            ch[p] = true
            answer += "\(p) "
            for c in graph[p] { queue.append(c) }
        }
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
    static boolean[] ch;
    static Map<Integer, ArrayList<Integer>> map = new HashMap<>();;
    static StringBuilder sb = new StringBuilder();
    
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        
        int N = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());
        int V = Integer.parseInt(st.nextToken());
        
        for(int i=1; i<=N; i++) map.put(i, new ArrayList<Integer>());

        for(int i=0; i<M; i++) {
            st = new StringTokenizer(br.readLine());
            int u = Integer.parseInt(st.nextToken());
            int v = Integer.parseInt(st.nextToken());

            ArrayList<Integer> list = map.get(u);
            list.add(v);
            map.put(u, list);
            
            list = map.get(v);
            list.add(u);
            map.put(v, list);
        }

        ch = new boolean[N+1];
        dfs(V);
        sb.append("\n");
        
        ch = new boolean[N+1];
        bfs(V);
        
        System.out.println(sb);
    }

    public static void dfs(int cur) {
        ch[cur] = true;
        sb.append(cur).append(" ");

        List<Integer> nodes = new ArrayList<>(map.get(cur));
        Collections.sort(nodes);
        for(int n : nodes) {
            if(ch[n]) continue;
            dfs(n);
        }
    }

    public static void bfs(int start) {
        Queue<Integer> q = new LinkedList<>();
        q.offer(start);
        ch[start] = true;
        while(!q.isEmpty()) {
            int cur = q.poll();
            sb.append(cur).append(" ");

            List<Integer> nodes = new ArrayList<>(map.get(cur));
            Collections.sort(nodes);
            for(int n : nodes) {
                if(!ch[n]) {
                    q.offer(n);
                    ch[n] = true;
                }
            }
        }
    }
}